tensorflow/java/src/main/native/operation_builder_jni.cc

```diff
@@ -257,3 +257,32 @@ JNIEXPORT void JNICALL Java_org_tensorflow_OperationBuilder_setAttrShape(
   TF_SetAttrShape(d, cname, cvalue.get(), static_cast<int>(num_dims));
   env->ReleaseStringUTFChars(name, cname);
 }
+
+JNIEXPORT void JNICALL Java_org_tensorflow_OperationBuilder_setAttrStringList(
+        JNIEnv* env, jclass object, jlong handle, jstring name, jobjectArray values) {
+ static_assert(sizeof(jbyte) == 1,
+                "Require Java byte to be represented as a single byte");
+  TF_OperationDescription* d = requireHandle(env, handle);
+  if (d == nullptr) return;
+  const char *cname = env->GetStringUTFChars(name, nullptr);
+  int num_strings = env->GetArrayLength(values);
+  size_t *lengths = new size_t[num_strings];
+  jbyte **strings = new jbyte *[num_strings];
+  for (int i = 0; i < num_strings; i++) {
+    jbyteArray value = (jbyteArray) env->GetObjectArrayElement(values, i);
+    lengths[i] = static_cast<size_t>(env->GetArrayLength(value));
+    strings[i] = env->GetByteArrayElements(value, nullptr);
+  }
+  if (num_strings > 0) {
+    TF_SetAttrStringList(d, cname, reinterpret_cast<const void *const *>(strings), lengths, num_strings);
```

Comment:
The presence of this `reinterpret_cast`  and what I think is incorrect cleanup (see comments later on) makes me think that we should make this implementation more explicit and simpler. How about something like the following (do validate that this is correct :):

```
int num_values = env->GetArrayLength(values);
std::unique_ptr<jbyteArray[]> jarrays(new jbyteArray[num_values]);
std::unique_ptr<jbyte*[]> jvalues(new jbyte*[num_values]);
std::unique_ptr<void*[]> cvalues(new void*[num_values]);
std::unique_ptr<size_t[]> lengths(new size_t[num_values]);

for (int i = 0; i < num_values; ++i) {
  jByteArray v = static_cast<jbyteArray>(env->GetObjectArrayElement(values, i));
  jarrays[i] = v;
  jvalues[i] = env->GetByteArrayElements(v, nullptr);
  cvalues[i] = jvalues[i];
  lengths[i] = static_cast<size_t>(env->GetArrayLength(v));
}
TF_SetAttrStringList(d, cname, cvalues.get(), lengths.get(), num_values);
for (int i = 0; i < num_values; ++i) {
  env->ReleaseByteArrayElements(jarrays[i], jvalues[i], JNI_ABORT);
}
env->ReleaseStringUTFChars(name, cname);
```

I realize that this has additional allocations and overheads that could be optimized away. However, I don't think that this function is going to be common enough to be a performance bottleneck so the explicit clarity is well worth it. If, in the future the allocations are of a concern, we can optimize it, but for now we don't need to.